pkg/apis/openstackproviderconfig/v1alpha1/types.go

```diff
@@ -62,8 +67,24 @@ type RootVolume struct {
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// OpenstackClusterProviderConfig is the providerConfig for OpenStack in the cluster object
+// +k8s:openapi-gen=true
+type OpenstackClusterProviderConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// NodeCIDR is the OpenStack Subnet to be created
+	NodeCIDR string `json:"nodeCidr,omitempty"`
+	// Floating IP which will be assigned to a created loadbalancer. If not provided,
+	// no loadbalancer will be created.
+	FloatingIP string `json:"floatingIP,omitempty"`
```

Comment:
I wonder if we should pick a better name for this. I hate to nitpick on names but as a user, it might not be entirely obvious to me that setting a floatingIP ends up creating a load balancer. Thoughts? Maybe call it LoadBlanacerIP?

Reply:
What is it the FloatingIP/LoadBalancerIP for? I would be more specific, indicating that this is for access to the control plane (I think?)

Reply:
I will leave this out. Currently not sure how to implement it.

The rough idea is to create a floating IP for the loadbalancer. That then can be used for a HA setup. This IP must reach the first master anyways, because it also needs to include it in its SANs. To create the actual LB we might decide to create a Type-LB-Service for kube-apiserver. But I would leave this out of scope here.